mailbag/controller.py

```diff
@@ -112,4 +111,15 @@ def generate_mailbag(self):
                         writer = csv.writer(f)
                         writer.writerows(portion)
 
+        if self.args.compress and not self.args.dry_run:
+            compress=self.args.compress.lower()
+            if compress=="tar":
+                shutil.make_archive(mailbag_dir, 'tar', mailbag_dir)
+            elif compress== "zip":
+                shutil.make_archive(mailbag_dir, 'zip', mailbag_dir)
+            elif compress == "tar.gz":
+                shutil.make_archive(mailbag_dir, 'gztar', mailbag_dir)
+            else:
+                log.error("Invalid compression format")
+
```

Comment:
needs `shutil.rmtree(mailbag_dir) but only if a valid format was provided. Probably a good idea to test if the file exists first before deleting.